src/client/pluginChannels.js

```diff
@@ -87,7 +89,7 @@ module.exports = function (client, options) {
   function writeDumbArr (value, buf, offset) {
     // TODO: Remove trailing \0
     value.forEach(function (v) {
-      offset += this.write(v, buf, offset, 'cstring', {})
+      offset += buf.write(v, offset)
```

Comment:
this is a change of meaning for the value arg, it now assumes that value is a bugger, is that true?

Reply:
No, value is the array of strings

Reply:
ok so that code is not going to work then

Reply:
It's working and tested..

Reply:
`this.write` throw an error. I mean, it says `this.write is not a function`. Which is weird because if you do a `console.log(this.write)` above a function is correctly logged. But I'm sure that `buf.write` should be used here.

Reply:
this.write is indeed wrong because it's not a function
However buf.write is also wrong because it's for writing raw bytes

You need to use protodef to serialize that array of strings

Reply:
No actually pretty sure that this.write is defined, the this is the protodef instance

Reply:
> However buf.write is also wrong because it's for writing raw bytes

Unless you set the encoding

Reply:
> No actually pretty sure that this.write is defined, the this is the protodef instance

`this` refers to the protodef instance for some reason. But again `this.write` throws an error when called in `writeDumbArr`

Reply:
And `buf.write` should be used instead of `this.write` even if the method was working because the `writeDumbArr` is used to serialize the actual array

Reply:
https://nodejs.org/api/buffer.html#bufwritestring-offset-length-encoding 
Buffer.write takes a string as argument, not an array of strings

Reply:
It's not the array of strings passed inside the big.write it's the v variable (value.forEach)

Reply:
I see that's true
still missing the zero terminator

Reply:
The zero terminator is not needed, should I add it anyway?

Reply:
how do you know it's not needed?

Reply:
if there's no 0 termination then you need length prefixing
you would definitely need to adapt read and size of as well

Reply:
A null separated list of strings (for me) looks like this (hex)
```
63 68 61 6E 6E 65 6C 31 00 CA FE BA BE
<----------------------------------> <--> <--------------->
               Channel1.                     Channel 2
                                      Separator

There's no need for a terminator at the end

Reply:
Oh wait are you telling me that if i put two strings there's no separator?

Reply:
Ok I get it. I'll fix this when I'll be home. I'm still wondering how the previous code was working.

Reply:
Updated! Now the null separator is present